specification/dartLangSpec.tex

```diff
@@ -23130,156 +23146,238 @@ \subsection{Operator Precedence}
-and we may hence just as well omit the initial structural traversal
-which might take many steps only to report that two large type terms
-are not quite identical.%
-}
-
-\LMHash{}%
-The rules are ordered by means of their rule numbers:
-A rule given here numbered $N.1$ is inserted immediately after rule $N$,
-followed by rule $N.2$, and so on,
-followed by the rule whose number is $N+1$.
-\commentary{%
-So the order is
-\AppSrnReflexivity, \SrnTop--\SrnTypeVariableReflexivityA,
-\AppSrnTypeVariableReflexivityB, \AppSrnTypeVariableReflexivityC,
-\AppSrnTypeVariableReflexivityD,
-\SrnRightPromotedVariable, and so on.%
-}
-
-\LMHash{}%
-We now specify the procedure which is used to determine whether
-\SubtypeStd{S}{T} holds,
-for some specific types $S$ and $T$:
-Select the first rule $R$ whose syntactic constraints are satisfied
-by the given types $S$ and $T$,
-and proceed to show that its premises hold.
-If so, we terminate and conclude that the subtype relationship holds.
-Otherwise we terminate and conclude
-that the subtype relationship does not hold,
-except if $R$ is
-\SrnRightFutureOrA, \SrnRightFutureOrB,
-\AppSrnRightFutureOrC, or \AppSrnRightFutureOrD.
-\commentary{%
-In particular, for the original query \SubtypeStd{S}{T},
-we do not backtrack into trying to use a rule that has
-a higher rule number than that of $R$,
-except that we may try all of
-the rules with \code{FutureOr<$T$>} to the right.%
-}
-
-\commentary{%
-Apart from the fact that the full complexity of subtyping
-is potentially incurred each time it is checked whether a premise holds,
-the checks applied for each rule is associated with an amount of work
-which is constant for all rules except the following:
-First, the group of rules
-\SrnRightFutureOrA, \SrnRightFutureOrB,
-\AppSrnRightFutureOrC, and \AppSrnRightFutureOrD{}
-may cause backtracking to take place.
-Next, rules \SrnPositionalFunctionType--\SrnCovariance{}
-require work proportional to the size of $S$ and $T$,
-due to the number of premises that must be checked.
-Finally, rule~\SrnSuperinterface{} requires work
-proportional to the size of $S$,
-and it may also incur the cost of searching up to the entire set of
-direct and indirect superinterfaces of the candidate subtype $S$,
-until the corresponding premise for one of them is shown to hold,
-if any.
-
-Additional optimizations are applicable.
-For instance,
-we can immediately conclude that the subtype relationship does not hold
-when we are about to check rule~\SrnSuperinterface{}
-if $T$ is a type variable or a function type.
-For several other forms of type, e.g.,
-a promoted type variable,
-\code{Object}, \DYNAMIC, \VOID,
-\code{FutureOr<$T$>} for any $T$, or \FUNCTION,
-it is known that it will never occur as $T$ for rule~\SrnSuperinterface,
-which means that this seemingly expensive step can be confined to some extent.%
-}
+  \commentary{%
+    Note that this rule is admissible, and can be safely elided if desired.%
```

Comment:
Define "admissible". It's a term of art not used elsewhere in the specification, so we cannot assume the reader to be familiar with it.